pulp_2to3_migration/app/pre_migration.py

```diff
@@ -621,14 +635,23 @@ def delete_old_resources(plan):
     old_dist_query &= Q(pulp2_repository__in=repos_with_old_distributions_qs) | Q(not_in_plan=True)
 
     with transaction.atomic():
-        pulp2distributors_with_old_distributions_qs = Pulp2Distributor.objects.filter(
-            old_dist_query).only('pulp3_distribution')
-
+        pulp2distributors_with_old_distributions_qs = Pulp2Distributor.objects.filter(old_dist_query)
+        pb_to_delete = set()
+        dist_to_delete = []
         for pulp2distributor in pulp2distributors_with_old_distributions_qs:
             if pulp2distributor.is_migrated:
                 pulp2distributor.is_migrated = False
                 pulp2distributor.save()
             if pulp2distributor.pulp3_publication:
-                pulp2distributor.pulp3_publication.delete()
+                # check if publication is shared by multiple distributions
+                pulp2dists = pulp2distributor.pulp3_publication.pulp2distributor_set.all()
+                publication = pulp2distributor.pulp3_publication
```

Comment:
Why is the `publication` variable needed? and why here and not earlier since `pulp2distributor.pulp3_publication` is used twice before this line.
For consistency sake, I suggest either to remove it, or to have the similar variable for a distribution.

Reply:
sorry, i think these a leftovers from previous versions of the code, `publication` variable is not needed